modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java

```diff
@@ -1214,14 +1214,18 @@ private String toExampleValueRecursive(String modelName, Schema schema, Object o
             ArraySchema arrayschema = (ArraySchema) schema;
             Schema itemSchema = arrayschema.getItems();
             String itemModelName = getModelName(itemSchema);
-            if (objExample instanceof Iterable && itemModelName == null) {
-                // If the example is already a list, return it directly instead of wrongly wrap it in another list
-                return fullPrefix + objExample.toString() + closeChars;
-            }
             Set<Schema> newSeenSchemas = new HashSet<>(seenSchemas);
             newSeenSchemas.add(schema);
-            example = fullPrefix + "[" + "\n" + toExampleValueRecursive(itemModelName, itemSchema, objExample, indentationLevel + 1, "", exampleLine + 1, newSeenSchemas) + ",\n" + closingIndentation + "]" + closeChars;
-            return example;
+
+            if (objExample instanceof Iterable) {
+                String result = fullPrefix + "[" + "\n";
+                for (Object item : (Iterable) objExample) {
+                    result += toExampleValueRecursive(itemModelName, itemSchema, item, indentationLevel + 1, "", exampleLine + 1, newSeenSchemas) + ",\n";
+                }
+                result += closingIndentation + "]" + closeChars;
+                return result;
+            }
+            return fullPrefix + "[" + "\n" + toExampleValueRecursive(itemModelName, itemSchema, objExample, indentationLevel + 1, "", exampleLine + 1, newSeenSchemas) + ",\n" + closingIndentation + "]" + closeChars;
```

Comment:
A lot of heavy lifting is happening in this method. I feel that covering that by some unit tests could be beneficial. Any thoughts?